From 2527c37f956aa38408e251587556a6c8389af319 Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Sun, 29 Dec 2019 23:38:28 +0000
Subject: [PATCH] configure.ac: autodetect need for external libintl

uclibc-ng and macos standard libraries don't provide libintl
facilities. As a result build fails as
(x86_64-gentoo-linux-uclibc example):

```
...
gcc  -g -O2   -o rpcgen rpc_clntout.o rpc_cout.o rpc_hout.o rpc_main.o rpc_parse.o rpc_sample.o rpc_scan.o rpc_svcout.o rpc_tblout.o rpc_util.o
/usr/lib/gcc/x86_64-gentoo-linux-uclibc/9.2.0/../../../../x86_64-gentoo-linux-uclibc/bin/ld: rpc_main.o: in function `usage':
/root/rpcsvc-proto/rpcgen/rpc_main.c:1403: undefined reference to `libintl_gettext'
/usr/lib/gcc/x86_64-gentoo-linux-uclibc/9.2.0/../../../../x86_64-gentoo-linux-uclibc/bin/ld: /root/rpcsvc-proto/rpcgen/rpc_main.c:1404: undefined reference to `libintl_gettext'
...
```

The change is simple: add gettext m4 macros to detect need
for libintl and link to an external library if needed.

Tested on x86_64-glibc and x86_64-uclibc-ng.

Reported-by: Anton
Bug: https://bugs.gentoo.org/637084
Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>

--- a/configure.ac
+++ b/configure.ac
@@ -16,4 +16,7 @@ AC_ISC_POSIX
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
 
+AM_GNU_GETTEXT([external])
+AM_GNU_GETTEXT_VERSION([0.18.1])
+
 AC_OUTPUT([Makefile rpcgen/Makefile rpcsvc/Makefile])
--- a/rpcgen/Makefile.am
+++ b/rpcgen/Makefile.am
@@ -10,3 +10,4 @@ noinst_HEADERS = proto.h rpc_parse.h rpc_scan.h rpc_util.h
 rpcgen_SOURCES = rpc_clntout.c rpc_cout.c rpc_hout.c rpc_main.c \
 	rpc_parse.c rpc_sample.c rpc_scan.c rpc_svcout.c rpc_tblout.c \
 	rpc_util.c
+rpcgen_LDADD = $(LIBINTL)
-- 
2.24.1

